ehrql/query_engines/mssql.py

```diff
@@ -210,23 +270,35 @@ def get_results_stream(self, dataset):
                 )
 
                 for i, results_table in enumerate(results_tables):
+                    # fetch_table_in_batches needs a key column to sort on in order to
```

Comment:
Ah yeah, this is exactly what my new PR lets us address (if we add a new `QueryType.AGGREGATE` for these sorts of queries).